src/main/groovy/net/wasdev/wlp/gradle/plugins/Liberty.groovy

```diff
@@ -55,92 +62,114 @@ class Liberty implements Plugin<Project> {
         }
 
         project.task('libertyDebug', type: DebugTask, dependsOn: 'libertyCreate') {
             description 'Runs the Liberty Profile server in the console foreground after a debugger connects to the debug port (default: 7777).'
             logging.level = LogLevel.INFO
-            server = currentServer
+            server = project.liberty.server
         }
 
         project.task('deploy', type: DeployTask) {
             description 'Deploys a supported file to the WebSphere Liberty Profile server.'
             logging.level = LogLevel.INFO
-            server = currentServer
+            server = project.liberty.server
         }
 
         project.task('undeploy', type: UndeployTask) {
             description 'Removes an application from the WebSphere Liberty Profile server.'
             logging.level = LogLevel.INFO
-            server = currentServer
+            server = project.liberty.server
         }
 
         project.task('installFeature', type: InstallFeatureTask, dependsOn: 'installLiberty') {
             description 'Install a new feature to the WebSphere Liberty Profile server'
             logging.level = LogLevel.INFO
-            server = currentServer
+            server = project.liberty.server
         }
         project.task('uninstallFeature', type: UninstallFeatureTask, dependsOn: 'installLiberty') {
             description 'Uninstall a feature from the WebSphere Liberty Profile server'
             logging.level = LogLevel.INFO
-            server = currentServer
+            server = project.liberty.server
         }
 
         project.task('cleanDirs', type: CleanTask) {
             description 'Deletes files from some directories from the WebSphere Liberty Profile server'
             logging.level = LogLevel.INFO
-            server = currentServer
+            server = project.liberty.server
         }
 
         project.task('installApps', type: InstallAppsTask, dependsOn: 'libertyCreate') {
             description "Copy applications generated by the Gradle project to a Liberty server's dropins or apps directory."
             logging.level = LogLevel.INFO
-            server = currentServer
+            server = project.liberty.server
+        }
+    }
+
+    ServerExtension copyProperties(LibertyExtension liberty) {
+        def serverMap = new ServerExtension().getProperties()
+        def libertyMap = liberty.getProperties()
+
+        serverMap.keySet().each { String element ->
+            if (element.equals("name")) {
+                serverMap.put(element, libertyMap.get("serverName"))
+            }
+            else {
+                serverMap.put(element, libertyMap.get(element))
+            }
         }
+        serverMap.remove('class')
+
+        return ServerExtension.newInstance(serverMap)
     }
 
+    void setServersForTasks(Project project){
+        project.tasks.withType(AbstractServerTask).each {task ->
+            task.server = project.liberty.server
```

Comment:
This is going to be problematic when we introduce the named domain objects.  Without some changes here all our named server tasks will end up with the same default server object.  Maybe just add a TODO for now that we need to beware.